tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1094,6 +1098,70 @@ const reducer = createReducer(
       singleSelectionHeaders: newOrder,
     };
   }),
+  on(actions.dataTableColumnEdited, (state, {fobState, newHeaders}) => {
+    const enabledNewHeaders: ColumnHeader[] = [];
+    const disabledNewHeaders: ColumnHeader[] = [];
+
+    newHeaders.forEach((header) => {
+      if (header.enabled) {
+        enabledNewHeaders.push(header);
+      } else {
+        disabledNewHeaders.push(header);
+      }
+    });
+
+    if (fobState === FobState.RANGE) {
+      return {
+        ...state,
+        rangeSelectionHeaders: enabledNewHeaders.concat(disabledNewHeaders),
+      };
+    }
+
+    return {
+      ...state,
+      singleSelectionHeaders: enabledNewHeaders.concat(disabledNewHeaders),
+    };
+  }),
+  on(actions.dataTableColumnToggled, (state, {fobState, headerType}) => {
+    const targetedHeaders =
+      fobState === FobState.RANGE
+        ? state.rangeSelectionHeaders
+        : state.singleSelectionHeaders;
+
+    const toggledHeaderCurrentIndex = targetedHeaders.findIndex(
+      (element) => element.type === headerType
+    );
+
+    // If the header is being enabled it goes at the bottom of the currently
+    // enabled headers. If it is being disabled it goes to the top of the
+    // currently disabled headers.
+    let newToggledHeaderIndex = getEnabledCount(targetedHeaders);
+    if (targetedHeaders[toggledHeaderCurrentIndex].enabled) {
+      newToggledHeaderIndex--;
```

Comment:
Nit about naming consistency. Maybe choose one of the following pairs:

currentToggledHeaderIndex/newToggledHeaderIndex

or

toggledHeaderCurrentIndex/toggledHeaderNewIndex

Reply:
Makes sense. I chose currentToggledHeaderIndex/newToggledHeaderIndex